### PR TITLE
MailService enhanced to send emails using SendGrid templates

### PR DESCRIPTION
**Pull request by wedge76**

### CODE REVIEW

File: Supertext.Base/Net/Mail/EmailInfoTemplates.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+
+namespace Supertext.Base.Net.Mail
+{
+    public class EmailInfoTemplates
+    {
+        public EmailInfoTemplates(string templateId, string dynamicTemplateDataAsJson, PersonInfo from, PersonInfo to,
+                                  string subject)
```
Comment (jameson-supertext) [inline Supertext.Base/Net/Mail/EmailInfoTemplates.cs:8]: Why is only parameter on its own line? Either they should all be on the same line, or they should all be their own lines.

Comment (wedge76) [reply Supertext.Base/Net/Mail/EmailInfoTemplates.cs:11]: fixed

File: Supertext.Base.Net/Mail/MailService.cs [hunk 2]
```diff
@@ -44,6 +47,39 @@ await SendInternalAsync(mail,
...
+                };
+                var client = new SendGridClient(options);
+                var message = new SendGridMessage
+                              {
+                                  TemplateId = mailInfo.TemplateId,
+                                  From = new EmailAddress(mailInfo.From.Email, mailInfo.From.Name),
+                                  Subject = mailInfo.Subject
+                              };
+                message.SetTemplateData(mailInfo.DynamicTemplateDataAsJson);
+                message.AddTos(mailInfo.Recipients.Select(r => new EmailAddress(r.Email, r.Name)).ToList());
```
Comment (jameson-supertext) [inline Supertext.Base.Net/Mail/MailService.cs:66]: Just an idea: How about a static extension method which converts `PersonInfo` to `EmailAddress`? Could be used on lines 62 and 66.

Comment (wedge76) [reply Supertext.Base.Net/Mail/MailService.cs]: refined

**Review by jameson-supertext — APPROVED** [review]

**Review by jameson-supertext — APPROVED** [review]

**Pull request merged**